e2e/e2e_test.go

```diff
@@ -87,17 +108,202 @@ var _ = Describe("Whereabouts functionality", func() {
+
+					// Increase to desired number of replicas for test
+					allPods, err := clientInfo.Client.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					maxReplicas := (envVars.numComputeNodes*maxPodsPerNode - (len(allPods.Items))) * envVars.fillPercentCapacity / 100
+					replicaCount = int32(maxReplicas)
+					replicaSet, err = clientInfo.updateReplicaSet(replicaSetObject(
+						replicaCount,
+						podTierLabel(rsName),
+						podNetworkSelectionElements(testNetworkName),
+					))
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(WaitForReplicaSetSteadyState(clientInfo.Client, testNamespace, label, replicaSet, rsSteadyTimeout)).To(Succeed())
+
+					Expect(iPPoolConsistency(envVars.kubeconfigPath, k8sClient, clientInfo.Client)).To(Succeed())
+				}
+			})
+		})
 	})
 })
 
-func clusterConfig() (*rest.Config, error) {
+func iPPoolConsistency(kubeconfigPath string, k8sClient *kubeClient.Client, cs *kubernetes.Clientset) error {
+	By("checking if there are any stale IPs in IP pool or any IPs in IP pool that are not seen attached to a pod")
+
+	By("Forcing reconciliation of the cluster...")
+	if err := runIpReconciler(kubeconfigPath); err != nil {
+		return err
+	}
+
+	By("Comparing pod ip assignments to ippool reservations")
+	// TODO - stop this ipPools[0] stuff and get it by name if there's a way to do it. low priority though
+	ipPools, err := k8sClient.ListIPPools(context.Background())
+	if err != nil {
+		return err
+	}
+
+	ipPoolAllocations := ipPools[0].Allocations() // todo, get this more elegantly than ipPools[0]
+	podList, err := ListPods(cs, testNamespace, wbLabelEqual)
+	if err != nil {
+		return err
+	}
+
+	found := true
+	podIP := ""
+	reservedIP := ""
+
+	for _, allocation := range ipPoolAllocations {
+		found = false
+		for _, pod := range podList.Items {
+			podIP, err = secondaryIfaceIPValue(&pod)
+			if err != nil {
+				return err
+			}
+			reservedIP = allocation.IP.String()
+
+			if reservedIP == podIP {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("Possible stale IP Pool: failed to find pod for IP Pool IP %s\n", allocation.IP)
+		}
+	}
+
+	for _, pod := range podList.Items {
+		found = false
+		for _, allocation := range ipPoolAllocations {
+			podIP, err = secondaryIfaceIPValue(&pod)
+			if err != nil {
+				return err
+			}
+			reservedIP = allocation.IP.String()
+
+			if reservedIP == podIP {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			fmt.Printf("possible pod IP not recorded in IP pool: failed to find IP pool allocation for pod IP %s\n", podIP)
+			return fmt.Errorf("possible pod IP not recorded in IP pool: failed to find IP pool allocation for pod IP %s\n", podIP)
+		}
+	}
+
+	return nil
+}
+
+type envVars struct {
+	kubeconfigPath      string
+	numComputeNodes     int
+	fillPercentCapacity int
+	numThrashIter       int
+}
+
+func getEnvVars() (envVars, error) {
```

Comment:
please avoid the `get` function name prefix - you oould name this `environment`, for instance.